src/OrchardCore.Modules/OrchardCore.Layers/Services/DefaultLayersMethodProvider.cs

```diff
@@ -9,48 +9,42 @@ namespace OrchardCore.Layers.Services
 {
     public class DefaultLayersMethodProvider : IGlobalMethodProvider
     {
-        private readonly GlobalMethod _isHomepage;
-        private readonly GlobalMethod _isAnonymous;
-        private readonly GlobalMethod _isAuthenticated;
-        private readonly GlobalMethod _url;
-        private readonly GlobalMethod _culture;
-
         private readonly GlobalMethod[] _allMethods;
 
         public DefaultLayersMethodProvider()
         {
-            _isHomepage = new GlobalMethod
+            var isHomepageMethod = new GlobalMethod
             {
                 Name = "isHomepage",
-                Method = serviceprovider => (Func<string, object>)(name =>
+                Method = serviceProvider => (Func<string, object>)(name =>
                 {
-                    var httpContext = serviceprovider.GetRequiredService<IHttpContextAccessor>().HttpContext;
+                    var httpContext = serviceProvider.GetRequiredService<IHttpContextAccessor>().HttpContext;
                     var requestPath = httpContext.Request.Path;
-                    return requestPath == "/" || string.IsNullOrEmpty(requestPath);
+                    return BooleanBoxes.Box(requestPath == "/" || string.IsNullOrEmpty(requestPath));
                 })
             };
 
-            _isAnonymous = new GlobalMethod
+            var isAnonymousMethod = new GlobalMethod
             {
                 Name = "isAnonymous",
-                Method = serviceprovider => (Func<string, object>)(name =>
+                Method = serviceProvider => (Func<string, object>)(name =>
                 {
-                    var httpContext = serviceprovider.GetRequiredService<IHttpContextAccessor>().HttpContext;
-                    return !httpContext.User?.Identity.IsAuthenticated;
+                    var httpContext = serviceProvider.GetRequiredService<IHttpContextAccessor>().HttpContext;
+                    return BooleanBoxes.Box(!httpContext.User?.Identity.IsAuthenticated);
                 })
             };
-            
-            _isAuthenticated = new GlobalMethod
+
+            var isAuthenticatedMethod = new GlobalMethod
             {
                 Name = "isAuthenticated",
-                Method = serviceprovider => (Func<string, object>)(name =>
+                Method = serviceProvider => (Func<string, object>)(name =>
                 {
-                    var httpContext = serviceprovider.GetRequiredService<IHttpContextAccessor>().HttpContext;
-                    return httpContext.User?.Identity.IsAuthenticated;
+                    var httpContext = serviceProvider.GetRequiredService<IHttpContextAccessor>().HttpContext;
+                    return BooleanBoxes.Box(httpContext.User?.Identity.IsAuthenticated);
```

Comment:
Is this a performance optimization to avoid boxing? Is the performance improvement worth the added code complexity?

Reply:
Yeah it stops boxing booleans for each layer rule execution. I saw it as pretty simple, but happy to remove. Not a huge perf change as it is only a single object allocation saved per layer rule execution.

Reply:
Not a fan at all, for the same reasons as when I see `(object is null)`, no actual impact on the perf at our level, but more code to maintain and we don't do it everywhere. OC has a huge issue with allocations though, we'll have to look into it.

Reply:
removed the boolean boxes code.